src/git-rename-branch

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+source "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/helpers/helpers.sh"
+
+
+function preconditions {
+  target_branch_name=$1
+  destination_branch_name=$2
+  force_parameter=$3
+
+  force_rename=false
+  if [ "$force_parameter" == "-f" ]; then
+    force_rename=true
+  fi
+
+  if [ "$(is_main_branch "$target_branch_name")" = true ]; then
+    echo_error_header
+    echo_error "The main branch cannot be renamed."
+    exit_with_error newline
+  fi
+
+  if [ -z "$target_branch_name" ]; then
+    echo_error_header
+    echo_error "No branch name provided to rename."
+    exit_with_error newline
+  fi
+
+  if [ "$target_branch_name" != "$INITIAL_BRANCH_NAME" ]; then
+    ensure_has_branch "$target_branch_name"
+  fi
+
+  echo $force_rename
+  if [ "$force_rename" = false ]; then
+    ensure_is_feature_branch "$target_branch_name" "Run 'git rename-branch $target_branch_name $destination_branch_name -f' to force the rename, then reconfigure git-town on any other clones of this repo."
+  fi
+
+  if [ -z "$destination_branch_name" ]; then
```

Comment:
I think the errors should be different to be clear that two branch names are required, but I can move it up closer to the other.